### ISSUE TITLE
importer: add Instagram importer

### ISSUE THREAD

**Issue by bradfitz** [perkeep/perkeep#1126]
Tracking bug for instagram importer.

**Comment by plural** [comment]
is anyone already working on the instagram importer?  I'd like to help if i can.

**Comment by bradfitz** [comment]
@plural, nope. Please do!

Note that using the new-ish `pk-devimport` makes importer development a lot more fun & iterative. (faster hack/save/compile/run cycle)

**Comment by plural** [comment]
fantastic.  I've written some stuff against the instagram API in python a while back.  The instagram API docs talk about non-automated uses, so i am a bit worried we will need to prevent it from running scheduled, but i'll dig in more and see what's up.

**Comment by robertgzr** [comment]
@plural not sure how far you got but it seems like instagram has seriously limited their public API https://www.instagram.com/developer/changelog and they plan on deprecating reading the authenticated users profile and media until 2020 :/

**Comment by robertgzr** [comment]
@bradfitz @mpl 
I have a working instagram importer on top of [goinsta](https://github.com/ahmdrz/goinsta) but I'm a bit stuck on how to test it... I would just submit what I have for review :p 

I also haven't really looked into the ui part too much...

**Comment by mpl** [comment]
@robertgzr sure, if it seems to be working, that's a good enough starting point for a CL. Send away please.

For an integration test, you can look at what's already been done in other importers, such an the pinboard one.

**Comment by mpl** [comment]
@robertgzr I've started reviewing, but before I go further, can you please explain better why we need to use goinsta at all? In particular, it bothers me that we would have to ask the user's credentials (username and password) while the API seems to be offering good old OAuth2: https://www.instagram.com/developer/authentication/

Feel free to point at parts of the code directly in the CL (even the vendored parts) to explain better.

**Comment by mpl** [comment]
@robertgzr yes, I had a look at the endpoints, and I feared the 20 entries limitation, but good to know you've actually tried.

But then comes the next question (because I'm hoping you've had a look at the goinsta code, I haven't yet): the goinsta devs do not have anymore access to a broader instagram API than we do, so how do they do it? And is the way they do it complicated enough that we have to import all of goinsta instead of redoing it ourselves for our (I assume) simpler needs?

**Comment by robertgzr** [comment]
@mpl They reverse-engineered the "non-public" instagram api (what the app is using I assume), probably similar to https://github.com/mgp25/Instagram-API (which has quite elaborate technical documentation) by basically emulating the app.

I wouldn't want to replicate all that stuff... but I see what you mean. The alternative would be to have this be it's own app instead of an integrated importer.

**Comment by mpl** [comment]
so they're basically using undocumented endpoints?

**Comment by mpl** [comment]
https://perkeep-review.googlesource.com/c/perkeep/+/17606
